Remove cpu limits for dns pod to avoid CPU starvation

vishh opened the pull request:
The current limits are not based on usage profiles
Fixes #33222

``` release-note
Remove cpu limits for dns pod to avoid CPU starvation
```

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33227)

<!-- Reviewable:end -->

vishh committed:
remove cpu limits for dns pod. The current limits are not based on usage profiles

Signed-off-by: Vishnu kannan <vishnuk@google.com>

bprashanth commented on cluster/addons/dns/skydns-rc.yaml.base:
```diff
@@ -103,9 +101,7 @@ spec:
         resources:
-          # keep request = limit to keep this container in guaranteed class
           limits:
-            cpu: 10m
```
this is guaranteed qos, so what happens now?

vishh replied:
As of v1.4 this pod was in the Burstable QoS class even before this patch. QoS is per pod and per container.

bprashanth replied:
oh, hmm, ok thinking

bprashanth replied:
Please add a comment saying that the pod is anyway in "burstable" because of memory, so cpu limits don't matter. Also include that we *don't* want to set cpu until we've profiled the sub components thoroughly, because on debian systems the enforcement is more strict.

vishh replied:
Adding more TODO seems to be making the yaml outputs incompatible. Given the time constraint, I chose to not do that. I'm happy to post another PR tomorrow once I have more time to figure out why the yaml outputs were broken when TODOs are being added.

bprashanth replied:
if you confirm comments are what broke the make-ing, please file a bug

pwittrock commented:
@k8s-bot smoke test this

pwittrock commented:
@k8s-bot gke test this

vishh commented:
@k8s-bot gce e2e test this

vishh commented:
@k8s-bot gke smoke test this

vishh commented:
@k8s-bot kubemark test this

vishh commented:
@k8s-bot gke e2e test this

bprashanth commented:
LGTM when you fix the nits, I assume you used the makefile to generate the yamls from the one base yaml. Please also answer https://github.com/kubernetes/kubernetes/issues/33222#issuecomment-248774782 so we have a reference of the impact of the change, if we need to debug a user issue.

vishh commented:
I think I fixed the issues with the config files. Turns out the `TODO` line that I added was confusing the yaml format. I removed it for now. 
If tests pass this time around, this should be good to go @pwittrock

jessfraz commented:
was cherry-picked in #34266
